spanishdict/pn-logging
# Chore upgrade eslint

@ahaurw01 Let's kick this off slowly. https://www.pivotaltracker.com/story/show/110815626

In this PR:
1. Updated eslint rules to https://github.com/spanishdict/playbook/blob/master/code-quality/eslintrc-node0.10.json
2. Disabled rules that `index.js` violates. Don't want to touch source code now.

Final goal: 
1. Add sentry to pn-logging
2. Stop depending on config module, provide API to pass in config

I plan to do this in below order in several PRs:
1. Upgrade winston to latest without changing current API. Looking up API doc for current version doesn't provide us value. In order to use certain transporter, e.g. sentry, we have to upgrade anyway.
2. 100% tests coverage
3. Remove dependency on config
4. Explicit exports, no `module.exports[level] = _logger(level);`
5. Add sentry transport or just sentry

The final API should remain the same when possible.

/cc @danpaz
